src/pybind/mgr/cephadm/tests/test_cephadm.py

```diff
@@ -2261,3 +2262,33 @@ def test_inventory_known_hostnames(self, cephadm_module):
         assert cephadm_module.inventory.get_addr('host5') == '1.2.3.5'
         with pytest.raises(OrchestratorError):
             cephadm_module.inventory.get_addr('host5.domain')
+
+    def test_async_timeout_handler(self, cephadm_module):
+        cephadm_module.default_cephadm_command_timeout = 900
+
+        async def _timeout():
+            raise asyncio.TimeoutError
+
+        with pytest.raises(OrchestratorError, match=r'Command timed out \(default 900 second timeout\)'):
+            with cephadm_module.async_timeout_handler():
+                cephadm_module.wait_async(_timeout())
+
+        with pytest.raises(OrchestratorError, match=r'Command timed out on host hostA \(default 900 second timeout\)'):
+            with cephadm_module.async_timeout_handler('hostA'):
+                cephadm_module.wait_async(_timeout())
```

Comment:
Just to clarify: are we really testing something here rather than passing the parameters?

Reply:
just that the error messages end up formatted properly. It's the matching of the string against the error messages that's of value (since all the `async_timeout_handler` does is format error messages anyway)